easybuild/easyconfigs/g/GULP/gulp-6.0-fftw-blas-linking-gcc-9.3.patch

```diff
@@ -0,0 +1,52 @@
+Removing hard-coded paths in the mkgulp file, which is creating the makefile
+author: J. Sassmannshausen (NHS/GSTT)
+diff --git a/gulp-6.0.org/Src/mkgulp b/gulp-6.0/Src/mkgulp
+index 28afb80..84549b0 100755
+--- a/gulp-6.0.org/Src/mkgulp
++++ b/gulp-6.0/Src/mkgulp
+@@ -170,7 +170,7 @@ fi
+ #  Include file directories
+ #
+ #--USER--Start
+-echo 'INC=-I.. -I/usr/local/include/' >> makefile
++echo 'INC=-I.. -I${EBROOTFFTW}/include -I/usr/local/include/' >> makefile
```

Comment:
Remove `-I/usr/local/include` and switch to use the `FFTW_INC_DIR` instead

Reply:
This is where I am stuck: In the patch file I have:
```diff
-echo 'INC=-I.. -I/usr/local/include/' >> makefile
+echo 'INC=-I.. -I${FFTW_INC_DIR}' >> makefile
```

Which results in the makefile to:
`INC=-I.. -I${FFTW_INC_DIR}`
which translates in the build to:
```
gfortran  -O3 -ffpe-summary=invalid,zero,overflow  -I.. -I    -DFLUSH -DNOFOX     -DFFTW3  -c ../datatypes.F90
f951: Warning: Nonexistent include directory -DFLUSH [-Wmissing-include-dirs]
```
and the build fails. I was already playing around with that which is why I fell back to the environment variables and the hard-coded lib-name. 
I perfectly agree, that is not desirable, but I either miss out something here, or that code does things differently. 
There is no `configure` involved here! You run the `mkgulp` file which is creating the `makefile` .

Reply:
I can only assume that environment variables at all are expended in this context, since you were using them I assumed they were. Now I assume your previous `-I${EBROOTFFTW}/include` etc. just got expanded to `-I/include` and things just happened to work anyway because we set CPATH and LIBRARY_PATH which the compiler can see. If so, I would confirm if this is the case, and then investigate why all our (important) environments are dropped.